src/dune_rules/warning.mli

```diff
@@ -0,0 +1,26 @@
+(** General warning mechanism for dune rules *)
+
+open Import
+
+type t
+
+val create
```

Comment:
Can you add a comment explaining the difference between `default` and `since`. I understand it now after reading the code but I was confused at first.